app/lib/presentation/pages/home_page.dart

```diff
@@ -40,4 +55,28 @@ class HomePage extends StatelessWidget {
           builder: (context) => CustomCollectionsPage(),
         ),
       );
+  
+  void practiceCollectionsClicked(BuildContext context) async {
+    final fileContent = await rootBundle.loadString('assets/words/polish_words.txt');
```

Comment:
.txt file is not a good way to save such data.
A non-changing list can be `const`, which will improve performance.
Also it will be easier to work with.

And please move app logic like getting words of a language to the "backend" side of the app which is the infrastructure folder.
All of the collection should get returned as one object from there.